### PR TITLE
Annotation and user subviews

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: views/default/annotation/elements/summary/content.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+$content = elgg_extract('content', $vars, '');
+if ($content === false) {
+	return;
+}
+
+$annotation = elgg_extract('annotation', $vars);
+if ($content === '' && $annotation instanceof ElggAnnotation) {
+	$content = $annotation->value;
+}
+
```
Comment (jdalsem) [inline views/default/annotation/elements/summary/content.php]: if content empty return

File: views/default/annotation/elements/summary/icon.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+$icon_size = elgg_extract('icon_size', $vars, 'small');
+$icon_entity = elgg_extract('icon_entity', $vars);
+if ($icon_entity instanceof ElggEntity) {
+	echo elgg_view_entity_icon($icon_entity, $icon_size);
+	return;
+}
+
+if ($icon !== '') {
```
Comment (jdalsem) [inline views/default/annotation/elements/summary/icon.php]: ?? maybe duplicate code?

File: views/default/annotation/elements/summary/subtitle.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+$subtitle = elgg_extract('subtitle', $vars, '');
+if ($subtitle === false) {
+	return;
+}
+
+$annotation = elgg_extract('annotation', $vars);
+if ($subtitle === '' && $annotation instanceof ElggAnnotation) {
+	$subtitle = '';
```
Comment (jdalsem) [inline views/default/annotation/elements/summary/subtitle.php]: not needed

File: views/default/annotation/elements/summary/title.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	$owner = $annotation->getOwnerEntity();
+	if (!$owner instanceof ElggEntity) {
+		return;
+	}
+	
+	$title = elgg_view('output/url', [
+		'text' => $owner->getDisplayName(),
+		'href' => $owner->getURL(),
+		'is_trusted' => true,
+	]);
```
Comment (jdalsem) [inline views/default/annotation/elements/summary/title.php]: wrap in h3? check with likes

File: views/default/user/elements/imprint.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * Displays information about the location and brief description of the user
```
Comment (jdalsem) [inline views/default/user/elements/imprint.php]: no it doesnt

File: views/default/user/elements/imprint/briefdescription.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ *                                      Default is ''
+ */
+
+$entity = elgg_extract('entity', $vars);
+if (!$entity instanceof ElggUser) {
+	return;
+}
+
+$briefdescription = elgg_extract('briefdescription', $vars);
+if (!isset($time)) {
```
Comment (jdalsem) [inline views/default/user/elements/imprint/briefdescription.php]: uh time/?

File: views/default/user/elements/imprint/contents.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Displays information about the author, the time and the access of the post
```
Comment (jdalsem) [inline views/default/user/elements/imprint/contents.php]: update text

File: views/default/user/elements/summary/content.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+$content = elgg_extract('content', $vars, '');
+if ($content === false) {
+	return;
+}
+
+$entity = elgg_extract('entity', $vars);
+if ($content === '' && $entity instanceof ElggUser) {
+	if (elgg_view_exists('user/status')) {
+		$content = elgg_view('user/status', $vars);
```
Comment (jdalsem) [inline views/default/user/elements/summary/content.php]: should we only pass entity?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: views/default/annotation/elements/summary/icon.php (hunk 2 above)
Comment (jeabakker) [reply views/default/annotation/elements/summary/icon.php]: fixed

File: views/default/annotation/elements/summary/content.php (hunk 1 above)
Comment (jeabakker) [reply views/default/annotation/elements/summary/content.php]: fixed

File: views/default/annotation/elements/summary/subtitle.php (hunk 3 above)
Comment (jeabakker) [reply views/default/annotation/elements/summary/subtitle.php]: fixed

File: views/default/annotation/elements/summary/title.php (hunk 4 above)
Comment (jeabakker) [reply views/default/annotation/elements/summary/title.php]: fixed

File: views/default/user/elements/imprint.php (hunk 5 above)
Comment (jeabakker) [reply views/default/user/elements/imprint.php]: fixed

File: views/default/user/elements/imprint/briefdescription.php (hunk 6 above)
Comment (jeabakker) [reply views/default/user/elements/imprint/briefdescription.php]: fixed

File: views/default/user/elements/imprint/contents.php (hunk 7 above)
Comment (jeabakker) [reply views/default/user/elements/imprint/contents.php]: fixed

File: views/default/user/elements/summary/content.php (hunk 8 above)
Comment (jeabakker) [reply views/default/user/elements/summary/content.php]: fixed

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: views/default/annotation/elements/imprint/contents.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Displays imprint contents
```
[the remark this answers is not in the document]
Comment (jeabakker) [reply views/default/annotation/elements/imprint/contents.php]: fixed

**Review by jdalsem — APPROVED** [review]

**Pull request merged**